cmd/cloudtrail/permission-denied.go

```diff
@@ -0,0 +1,117 @@
+package cloudtrail
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/cloudtrail"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	ctUtil "github.com/openshift/osdctl/cmd/cloudtrail/pkg"
+	ctAws "github.com/openshift/osdctl/cmd/cloudtrail/pkg/aws"
+	"github.com/openshift/osdctl/pkg/osdCloud"
+	"github.com/openshift/osdctl/pkg/utils"
+	"github.com/spf13/cobra"
+)
+
+var (
+	errorCode = ".*Client.UnauthorizedOperation.*"
+)
+
+type permissonOptions struct {
+	clusterID      string
+	startTime      string
+	printEventUrl  bool
+	printRawEvents bool
+}
+
+func newCmdPermissionDenied() *cobra.Command {
+	ops := &permissonOptions{}
+	permissionDeniedCmd := &cobra.Command{
+		Use:   "permission-denied",
+		Short: "Prints cloudtrail permission-denied events to console.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return ops.run()
+		},
+	}
+	permissionDeniedCmd.Flags().StringVarP(&ops.clusterID, "cluster-id", "C", "", "Cluster ID")
+	permissionDeniedCmd.Flags().StringVarP(&ops.startTime, "since", "", "1h", "Specifies that only events that occur within the specified time are returned.Defaults to 1h.Valid time units are \"ns\", \"us\" (or \"µs\"), \"ms\", \"s\", \"m\", \"h\".")
+	permissionDeniedCmd.Flags().BoolVarP(&ops.printEventUrl, "url", "u", false, "Generates Url link to cloud console cloudtrail event")
+	permissionDeniedCmd.Flags().BoolVarP(&ops.printRawEvents, "raw-event", "r", false, "Prints the cloudtrail events to the console in raw json format")
+	permissionDeniedCmd.MarkFlagRequired("cluster-id")
+	return permissionDeniedCmd
+}
+
+func (p *permissonOptions) run() error {
```

Comment:
**Suggestion (optional):** we could refactor even further, if you look at this function and `run` in `write-events.go` you will see that most of the logic is duplicate, minus the filtering. This could be deduplicated. 

It's fine to leave it as is for now, we can do this in a follow up PR or leave it as is.